Sources/MapboxNavigation/InstructionLabel.swift

```diff
@@ -12,48 +20,73 @@ open class InstructionLabel: StylableLabel, InstructionPresenterDataSource {
     // displayed. The bounds of `InstructionLabel` will be used if this view is unset.
     weak var viewForAvailableBoundsCalculation: UIView?
     var shieldHeight: CGFloat = 30
-    var imageRepository: ImageRepository = .shared
     var imageDownloadCompletion: (() -> Void)?
-    weak var instructionDelegate: VisualInstructionDelegate?
+    weak var instructionDelegate: VisualInstructionDelegate? {
+        didSet {
+            updateLabelAttributedText()
+        }
+    }
     
     var instruction: VisualInstruction? {
         didSet {
-            guard let instruction = instruction else {
-                text = nil
-                instructionPresenter = nil
-                return
-            }
-            let update: InstructionPresenter.ShieldDownloadCompletion = { [weak self] (attributedText) in
-                guard let self = self else { return }
-                self.attributedText = attributedText
-                self.imageDownloadCompletion?()
-            }
-            
-            let presenter = InstructionPresenter(instruction,
-                                                 dataSource: self,
-                                                 imageRepository: imageRepository,
-                                                 traitCollection: traitCollection,
-                                                 downloadCompletion: update)
-            
-            let attributed = presenter.attributedText()
-            attributedText = instructionDelegate?.label(self, willPresent: instruction, as: attributed) ?? attributed
-            instructionPresenter = presenter
+            updateLabelAttributedText()
+        }
+    }
+    
+    private func updateLabelAttributedText() {
+        guard let instruction = instruction else {
+            text = nil
+            return
         }
+        
+        let attributed = instructionDelegate?.label(self, willUpdate: instruction) ?? attributedString(for: instruction, with: SpriteRepository.init())
```

Comment:
Using `instructionDelegate` to generate the `attributedString` with its `SpriteRepository`, instead of storing a `SpriteRepository` instance inside each `InstructionLabel`, to reduce the memory.